controllers/upgrade.js

```diff
@@ -0,0 +1,45 @@
+const PouchDB = require('pouchdb-core');
+PouchDB.plugin(require('pouchdb-adapter-http'));
+
+const buildDbUrl = 'https://staging.dev.medicmobile.org/_couch/builds';
+const buildDb = new PouchDB(buildDbUrl);
+const targetDbUrl = process.env.COUCH_URL;
+const targetDb = new PouchDB(targetDbUrl);
+
+module.exports = (version, username) => {
+  console.log('upgrade()', `Upgrading to ${version}…`);
+
+  console.log('upgrade()', 'Fetching newDdoc…');
+  return buildDb
+    .get(version, { attachments:true })
+    .then(newDdoc => console.log('upgrade()', 'Fetched newDdoc.') || newDdoc)
+
+    .then(newDdoc => console.log('upgrade()', 'Fetching oldDdoc…') || newDdoc)
+    .then(newDdoc =>
+      targetDb.get('_design/medic')
```

Comment:
I've changed the api medic-api exposes to expect the following JSON body:
```json
{
  "build": {
    "namespace": "medic",
    "application": "medic",
    "version": "1.2.3-beta.4"
  }
}
```

(as an example). We'll error if you don't pass that namespace and application, but it allows us to expand out later and pull this code out of api if we want to.